src/test/java/org/hisp/dhis/rules/RuleEngineVariableNameTests.java

```diff
@@ -0,0 +1,545 @@
+                RuleDataValue.create( new Date(), "test_program_stage", "test_data_element_two", "male" ) ), "" );
+
+        RuleEvent ruleEvent3 = RuleEvent.create( "test_event3", "test_program_stage3",
+            RuleEvent.Status.ACTIVE, new Date(), new Date(), "", null, Arrays.asList(
+                RuleDataValue.create( new Date(), "test_program_stage", "test_data_element_one", "8" ),
+                RuleDataValue.create( new Date(), "test_program_stage", "test_data_element_two", "male" ) ), "" );
+
+        List<RuleEffect> ruleEffects = ruleEngineBuilder.events( Arrays.asList( ruleEvent1, ruleEvent2 ) ).build()
+            .evaluate( ruleEvent3 ).call();
+
+        assertThat( ruleEffects.size() ).isEqualTo( 1 );
+        assertThat( ruleEffects.get( 0 ).ruleAction() ).isEqualTo( ruleAction );
+    }
+
+    @Test
+    public void testMinValueIfVariableName()
+        throws Exception
+    {
+        RuleAction ruleAction = RuleActionDisplayKeyValuePair.createForFeedback(
+            "test_action_content", "d2:minValue(#{"+VARIABLE_NAME+"})" );
+        RuleVariable ruleVariableOne = RuleVariableNewestEvent.create(
+            VARIABLE_NAME, "test_data_element_one", RuleValueType.NUMERIC );
+
+        RuleVariable ruleVariableTwo = RuleVariableNewestEvent.create(
+            "test_var_two", "test_data_element_two", RuleValueType.TEXT );
+
+        Rule rule = Rule.create( null, null, "true", Arrays.asList( ruleAction ), "" );
+
+        RuleEngine.Builder ruleEngineBuilder = getRuleEngineBuilder( rule,
+            Arrays.asList( ruleVariableOne, ruleVariableTwo ) );
+
+        RuleEvent ruleEvent1 = RuleEvent.create( "test_event1", "test_program_stage1",
+            RuleEvent.Status.ACTIVE, new Date(), new Date(), "", null, Arrays.asList(
+                RuleDataValue.create( new Date(), "test_program_stage", "test_data_element_one", "5" ),
+                RuleDataValue.create( new Date(), "test_program_stage", "test_data_element_two", "male" ) ), "" );
+
+        RuleEvent ruleEvent2 = RuleEvent.create( "test_event2", "test_program_stage2",
+            RuleEvent.Status.ACTIVE, new Date(), new Date(), "", null, Arrays.asList(
+                RuleDataValue.create( new Date(), "test_program_stage", "test_data_element_one", "7" ),
+                RuleDataValue.create( new Date(), "test_program_stage", "test_data_element_two", "male" ) ), "" );
+
+        RuleEvent ruleEvent3 = RuleEvent.create( "test_event3", "test_program_stage3",
+            RuleEvent.Status.ACTIVE, new Date(), new Date(), "", null, Arrays.asList(
+                RuleDataValue.create( new Date(), "test_program_stage", "test_data_element_one", "8" ),
+                RuleDataValue.create( new Date(), "test_program_stage", "test_data_element_two", "male" ) ), "" );
+
+        List<RuleEffect> ruleEffects = ruleEngineBuilder.events( Arrays.asList( ruleEvent1, ruleEvent2 ) ).build()
```

Comment:
It seems that the "current" event(in this case ruleEvent3) is systematically not included in the context. I was expecting the context to be complete with all events, including the one counted as current. For calls to d2:minValue and all other functions that looks at values in multiple events, these should compare all context events as well as the current one, and return the minimum value across all these events.

Reply:
@zubaira can correct me if I am wrong, but I think the current event is added in the context at the moment the whole valueMap is built

Reply:
yes @enricocolasante that's right. If we add event/enrollment (which is subject to evaluation) into the context before evaluation then i think it will throw an exception.